ComplementaryScripts/modelCuration/AddCofactorIon.m

```diff
@@ -0,0 +1,216 @@
+    model.rxnNotes{rxnIndex} = 'NOTES:added for BiomassUpdate';
+end
+
+% add gene standard name for new genes
+fid = fopen('../../ComplementaryData/databases/SGDgeneNames.tsv');
+yeast_gene_annotation = textscan(fid,'%s %s','Delimiter','\t','HeaderLines',1);
+fclose(fid);
+
+geneIndex = zeros(1,1);
+for i = 1: length(model.genes)
+    geneIndex = strcmp(yeast_gene_annotation{1}, model.genes{i});
+    if sum(geneIndex) == 1 && ~isempty(yeast_gene_annotation{2}{geneIndex})
+        model.geneNames{i} = yeast_gene_annotation{2}{geneIndex};
+    else
+        model.geneNames{i} = model.genes{i};
+    end
+end
+
+% Add protein name for genes
+for i = 1:length(model.genes)
+    model.proteins{i} = strcat('COBRAProtein',num2str(i));
+end
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Introduce two PseudoRxn for cofactors and ions
+group = unique(data.groups);
+for i = 1:length(group)
+%get new metID for pseudo mets :ion and cofactor
+    metID  = ['s_' getNewIndex(model.mets) '[c]'];
+    model  = addMetabolite(model,metID,'metName',[group{i} ' [cytoplasm]']);
+    [~,index] = ismember(metID,model.mets);
+    model.metNotes{index}      = 'NOTES: added for BiomassUpdate';
+
+% Get coefficients for pseudo rxns: ion pseudoreaction, cofactor pseudoreaction
+    newps.rxnName = {[group{i},' pseudoreaction']};
+    index = strcmp(group{i},data.groups);
+    newps.metIDs = data.mets(index);
+    newps.metcoef = data.abundances(index)*-1;
+
+% introduce two pseudo mets ion [cytoplasm], cofactor [cytoplasm] into the reaction
+    newps.metIDs = [newps.metIDs;metID];
+    newps.metcoef = [newps.metcoef;1];
+
+    rxnID = ['r_' getNewIndex(model.rxns)];
+    [model,rxnIndex] = addReaction(model, rxnID,...
+                       'reactionName', char(newps.rxnName),...
+                       'metaboliteList',newps.metIDs,...
+                       'stoichCoeffList',newps.metcoef,...
+                       'reversible',false,...
+                       'checkDuplicate',1);
+    [EnergyResults,RedoxResults] = CheckEnergyProduction(model,{['r_' newID]},EnergyResults,RedoxResults);
+    [MassChargeresults] = CheckBalanceforSce(model,{['r_' newID]},MassChargeresults);
+    if isempty(rxnIndex)
+        rxnIndex = strcmp(model.rxns,['r_' newID]);
+    end
+    model.rxnNotes{rxnIndex} = 'NOTES:added for BiomassUpdate';
+end
+
+% Change original biomass equation
+UnusedMets = {'sulphate [cytoplasm]','riboflavin [cytoplasm]', 'heme a [cytoplasm]'};
+[~,UnusedMets_index] = ismember(UnusedMets,model.metNames);
+AddMets = {'cofactor [cytoplasm]','ion [cytoplasm]'};
+[~,AddMets_index] = ismember(AddMets,model.metNames);
+biomassRxn = 'r_4041';
+[~,biomassRxn_index] = ismember(biomassRxn,model.rxns);
+model.S(UnusedMets_index,biomassRxn_index) = 0;
+model.S(AddMets_index,biomassRxn_index) = -1;
+
+% Save model:
+model = rmfield(model,'grRules');
+%cd ..
+model = minimal_Y6(model);
```

Comment:
Same as before for line 213: `'minimal_Y6' is not found in the current folder or on the MATLAB path, but exists in: C:\Users\bensan\Documents\repos\yeast-GEM\ComplementaryScripts\modelCuration`

Reply:
Addressed.